### PR TITLE
Fix node names of submenu items across the editor

### LINKED ISSUE #84582: Error "Invalid node name for submenu, ..." on project load in editor (mono).
### Godot version

v4.2.beta5.mono.official [4c96e9676]

### System information

Windows 11 Home 64-bit - Godot v4.2.beta5.mono

### Issue description

Error:
```
Invalid node name for submenu, the following characters are not allowed:
. : @ / " %
```
  
This only happens in the mono version for all projects. This is also a regression bug and doesn't happen in `v4.2.beta4.mono`.

The high level cause of the bug is the `Project > Tools > C#` submenu that fails to load.
image

The low level cause is that the "C#" submenu PopupMenu doesn't have a validate node name (or any explicit node name in general). The name defaults to `@PopupMenu@17128` and the validated name becomes `_PopupMenu_17128`.

The error is due to the addition of validation in this commit:
#84183 
https://github.com/godotengine/godot/commit/137b25cf53f1a0761065eaff0ae2482458740369

I would fix it directly and make a pull request if I knew the proper solution. It seems the node name validation is inconsistent with the "default placeholder node names".

Possible fixes:
- Give the PopupMenu an explicit node name at `modules\mono\editor\GodotTools\GodotTools\GodotSharpEditor.cs` line 501
Example:
```cs
_menuPopup = new PopupMenu();
_menuPopup.Name = "CSharp";
```
- Rethink the validation in the aboved mention change: #84183 

(Note: New to bug reporting, please let me know if I could have made it more proper.)

### Steps to reproduce

- Open or create any project in v4.2.beta5.mono.official [4c96e9676]
- Read output log

### Minimal reproduction project

Empty Godot project.
[Empty.zip](https://github.com/godotengine/godot/files/13282638/Empty.zip)

### PR DESCRIPTION
**Pull request by YuriSizov**
Fixes https://github.com/godotengine/godot/issues/84582, a regression from https://github.com/godotengine/godot/pull/84183. Also removes some programmer remarks and fixes some docs.

There is still an open question whether this validation is excessive or not, whether we should allow generated names here or not. We might want that, which means we need a second version of `validate_node_name` which allows `@` specifically. But I would argue that we want the user to name their node here. And the case of the C# module was the only place where we forgot to do that.

### CODE REVIEW

File: editor/plugins/animation_blend_space_1d_editor.cpp [hunk 1]
```diff
@@ -796,15 +796,14 @@ AnimationNodeBlendSpace1DEditor::AnimationNodeBlendSpace1DEditor() {
 
 	error_label = memnew(Label);
 	error_panel->add_child(error_label);
-	error_label->set_text("hmmm");
```
Comment (akien-mga) [inline editor/plugins/animation_blend_space_1d_editor.cpp:799]: lol

File: modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs [hunk 2]
```diff
@@ -497,7 +497,10 @@ public override void _EnablePlugin()
 
             AddChild(new HotReloadAssemblyWatcher { Name = "HotReloadAssemblyWatcher" });
 
-            _menuPopup = new PopupMenu();
+            _menuPopup = new PopupMenu
+            {
+                Name = "CSharpTools",
+            };
```
Comment (akien-mga) [inline modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs:503]: I don't know what's the preferred style but I notice the `AddChild` function above does this inline:
```suggestion
            _menuPopup = new PopupMenu { Name = "CSharpTools" };
```

**Review by akien-mga — APPROVED** [review]
Looks good to me.

I was initially confused by the bug report / reason for enforcing a valid non-autogenerated name for `PopupMenu`, as usually all nodes in the end can do their work without hassle regardless of their name.

But looking at the API, I now understand that there's a two step process where `PopupMenu` must first be created, named, and added as child, for that name to then be passed as argument to `add_submenu_item`. Seems a bit weird to rely on names like this, but that's a pre-existing problem we won't solve here :)

File: modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs (hunk 2 above)
Comment (YuriSizov) [reply modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs:503]: Maybe it's supposed to be inlined when it's just one property? cc @raulsntos

**Comment by YuriSizov** [comment]
> Seems a bit weird to rely on names like this, but that's a pre-existing problem we won't solve here :)

Yeah, something that we should've fixed in 4.0, but alas...

File: modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs (hunk 2 above)
Comment (raulsntos) [reply modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs:503]: Yes, I think we tend to inline them when it's just one property and the line is not overly long. But we don't really enforce this so we're probably not consistent everywhere. I think it's fine either way.

**Comment by Shilo** [comment]
Looks good. Thank you!

> There is still an open question whether this validation is excessive or not, whether we should allow generated names here or not. We might want that, which means we need a second version of `validate_node_name` which allows `@` specifically. But I would argue that we want the user to name their node here.

As for the open question. I think both changes are appropriate: allowing `@UnamedNode...` for submenus as well as enforcing the naming standard.

The reason I feel validation should allow unnamed nodes is because there are still many nodes in the editor that are unnamed (such as the top & run bar nodes). The consistency and validation seems wavy in those areas.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**